### DISCUSSION TITLE
a few questions

### DISCUSSION THREAD

**Discussion by hiqsociety** [withcatai/node-llama-cpp#45] [Q&A] [answered]
1. just realised i'm not using gpu vram though i compiled with cuda. how to use the vram?

2. how to specify how many gpu layers to offload in cuda mode? i've compiled for cuda but i'm not sure where to adjust gpu layer offload settings.

3. how to make it accepts an input and output for api use like chatgpt etc? i dont mind input with output on command line

**Comment by giladgd** [comment, marked as answer]
1. `llama.cpp` supports some parameters you can configure to customize the behavior of CUDA.
You can read more about it [here](https://github.com/ggerganov/llama.cpp/tree/51a7cf5c6e490b2f51c82daa76c4ca4f8d845826#cublas) and then set the parameters you need as environment variables when you compile using `node-llama-cpp download --cuda` and when you run your code
2. Docs: [`gpuLayers`](https://withcatai.github.io/node-llama-cpp/types/LlamaModelOptions.html#:~:text=number%20of%20layers%20to%20store%20in%20VRAM)
3. Docs: [use a model as a chatbot](https://withcatai.github.io/node-llama-cpp/index.html#md:as-a-chatbot)
